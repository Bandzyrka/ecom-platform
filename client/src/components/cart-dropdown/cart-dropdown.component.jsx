import React from 'react'

import { connect } from 'react-redux'
import {selectCartItems} from '../../redux/cart/cart.selectors'

import { useNavigate } from "react-router-dom";

import { toggleCartHidden } from '../../redux/cart/cart.actions'
import CartItem from '../cart-item/cart-item.component'
import {CartDropdownContainer, CartDropdownItems, CheckoutButton, EmptyMessage} from './cart-dropdown.styles.jsx'


export const CartDropdown = ({ cartItems, dispatch }) =>{
    let navigate = useNavigate();
    return(
    <CartDropdownContainer>
        {
        cartItems.length ?
        (
            <CartDropdownItems> 
            {
                cartItems.map(cartItem => <CartItem key={cartItem.id} item={cartItem}/>)
            }
            </CartDropdownItems>
        )
        : <EmptyMessage> Your cart is empty </EmptyMessage>
        }
        <CheckoutButton onClick={() => {
                navigate('/checkout')
                dispatch(toggleCartHidden())
            }}> 
            Go to Checkout 
        </CheckoutButton>
    </CartDropdownContainer>
)}
const mapStateToProps = state => ({cartItems: selectCartItems(state)})
export default connect(mapStateToProps)(CartDropdown);