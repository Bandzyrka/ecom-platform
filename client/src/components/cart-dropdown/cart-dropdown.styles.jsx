import styled from 'styled-components'
import CustomButton from '../custom-button/custom-button.component'

export const CartDropdownContainer = styled.div`
  position: absolute;
  width: 240px;
  height: 340px;
  display: flex;
  flex-direction: column;
  padding: 20px;
  border: 1px solid black;
  background-color: white;
  top: 90px;
  right: 40px;
  z-index: 5;
`

export const CartDropdownItems = styled.div`
  height: 240px;
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  ::-webkit-scrollbar {
  width: 5px;
  }
  ::-webkit-scrollbar-track {
  background: #f1f1f1; 
  }
  ::-webkit-scrollbar-thumb {
  background: #888; 
  }
  ::-webkit-scrollbar-thumb:hover {
  background: #555; 
  }
  `

export const CheckoutButton = styled(CustomButton)`
  margin-top: auto;
  `
CheckoutButton.displayName = 'CartDropdownButton';

export const EmptyMessage = styled.span`  
  font-size: 20px;
  margin: auto;
`
EmptyMessage.displayName = 'EmptyMessageContainer';


  