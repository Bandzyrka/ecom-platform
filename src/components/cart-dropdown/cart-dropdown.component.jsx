import React from 'react'
import CustomButton from '../custom-button/custom-button.component'
import './cart-dropdown.styles.scss'
import { connect } from 'react-redux'
import CartItem from '../cart-item/cart-item.component'
import {selectCartItems} from '../../redux/cart/cart.selectors'
import { withRouter }from 'react-router-dom'
import { toggleCartHidden } from '../../redux/cart/cart.actions.js'

const CartDropdown = ({ cartItems, history, dispatch }) =>
(
    <div className="cart-dropdown">
        {
        cartItems.length ?
        (
            <div className="cart-items"> 
            {
                cartItems.map(cartItem => <CartItem key={cartItem.id} item={cartItem}/>)
            }
            </div>
        )
        : <span className="empty-message"> Your cart is empty </span>
        }
        <CustomButton onClick={() => {
                history.push('/checkout')
                dispatch(toggleCartHidden())
            }}> 
            Go to Checkout 
        </CustomButton>
    </div>
)
const mapStateToProps = state => ({cartItems: selectCartItems(state)})
export default withRouter(connect(mapStateToProps)(CartDropdown));