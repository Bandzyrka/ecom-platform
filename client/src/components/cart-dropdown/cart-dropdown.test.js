import React from 'react';
import { shallow } from 'enzyme';

import { CartDropdown } from './cart-dropdown.component';
import CartItem from '../cart-item/cart-item.component';


describe('CartDropdown component', () => {
  let wrapper;
  
  let mockDispatch;
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    
    mockDispatch = jest.fn();

    const mockProps = {
      cartItems: mockCartItems,

      dispatch: mockDispatch
    };

    wrapper = shallow(<CartDropdown {...mockProps} />);
  });

  it('should render CartDropdown component', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('should render an equal number of CartItem components as the cartItems prop', () => {
    expect(wrapper.find(CartItem).length).toEqual(mockCartItems.length);
  });

  it('should render EmptyMessageContainer if cartItems is empty', () => {
    const mockProps = {
      cartItems: [],
      dispatch: mockDispatch
    };

    const newWrapper = shallow(<CartDropdown {...mockProps} />);
    expect(newWrapper.exists('EmptyMessageContainer')).toBe(true);
  });
});