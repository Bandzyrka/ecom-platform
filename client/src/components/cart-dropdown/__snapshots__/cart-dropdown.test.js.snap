// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown component 1`] = `ShallowWrapper {}`;

exports[`CartDropdown should render cart dropdown 1`] = `ShallowWrapper {}`;
